### PR TITLE
feat(planning_evaluator): add modified goal deviation

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

Add metrics of deviation from the modified goal pose

## Tests performed

[
TIER IV internal scenario test](https://evaluation.tier4.jp/evaluation/reports/1a8b80d6-d0e3-5cd0-bf50-fe525dda5e81/tests/cb1630f0-dd61-5401-9011-8943a1b1ead9/55aa61e1-cec9-5db5-81d7-201d1de5e859?project_id=prd_jt)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: evaluator/planning_evaluator/src/metrics/deviation_metrics.cpp [hunk 1]
```diff
@@ -77,5 +77,26 @@ Stat<double> calcVelocityDeviation(const Trajectory & ref, const Trajectory & tr
   return stat;
 }
 
+Stat<double> calcPoseLongitudinalDeviation(const Pose & base_pose, const Pose & target_pose)
```
Comment (kyoichi-sugahara) [inline evaluator/planning_evaluator/src/metrics/deviation_metrics.cpp:80]: `const Pose & target_pose`
can be replaced with 
`const Point & target_point`?
if direction of target_pose is not used.

File: evaluator/planning_evaluator/src/planning_evaluator_node.cpp [hunk 2]
```diff
@@ -159,6 +154,51 @@ void PlanningEvaluatorNode::onTrajectory(const Trajectory::ConstSharedPtr traj_m
...
+
+  DiagnosticArray metrics_msg;
+  metrics_msg.header.stamp = now();
+  for (Metric metric : metrics_) {
+    const auto metric_stat = metrics_calculator_.calculate(
+      Metric(metric), modified_goal_ptr_->pose, ego_state_ptr_->pose.pose);
+    if (!metric_stat) {
+      continue;
+    }
+    metric_stats_[static_cast<size_t>(metric)].push_back(*metric_stat);
```
Comment (kyoichi-sugahara) [inline evaluator/planning_evaluator/src/planning_evaluator_node.cpp:188]: TODO(sugahara)
fix memory leak

File: evaluator/planning_evaluator/src/planning_evaluator_node.cpp [hunk 3]
```diff
@@ -137,18 +126,24 @@ DiagnosticStatus PlanningEvaluatorNode::generateDiagnosticStatus(
 
 void PlanningEvaluatorNode::onTrajectory(const Trajectory::ConstSharedPtr traj_msg)
 {
+  if (!ego_state_ptr_) {
+    return;
+  }
+
   auto start = now();
   stamps_.push_back(traj_msg->header.stamp);
```
Comment (kyoichi-sugahara) [inline evaluator/planning_evaluator/src/planning_evaluator_node.cpp:134]: TODO(suaghara)
fix memory leak

File: evaluator/planning_evaluator/src/metrics/deviation_metrics.cpp (hunk 1 above)
Comment (kosuke55) [reply evaluator/planning_evaluator/src/metrics/deviation_metrics.cpp]: thanks, fixed in https://github.com/autowarefoundation/autoware.universe/pull/3053/commits/31c8b82f24c5d381c9bf12581bdfeb6f8ff4cf50

Comment (kosuke55) [reply evaluator/planning_evaluator/src/metrics/deviation_metrics.cpp]: (and https://github.com/autowarefoundation/autoware.universe/pull/3053/commits/4ac91a03e16fd568295cb92b3f981dba27e5e7fc)

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM

**Comment by kosuke55** [comment]
fix and add tests in https://github.com/autowarefoundation/autoware.universe/pull/3053/commits/6d3b9e309ac99445ea494a705cf782c1f778993c.  checked it passed

**Pull request merged**